clarity_ext/repository/step_repository.py

```diff
@@ -121,23 +121,13 @@ def _wrap_artifacts(self, artifacts):
         for artifact in artifacts:
             yield self._wrap_artifact(artifact)
 
-    def update_objects(self, objects):
+    def update_artifacts(self, artifacts):
         """
         Updates each entry in objects to db, which must have the function
         updated_rest_resource
         """
-        update_queue = []
-        for obj in objects:
-            if type(obj) == Analyte:
-                original_analyte_from_rest = ApiArtifact(self.session.api, id=obj.id)
-                update_queue.append(obj.updated_rest_resource(original_analyte_from_rest, self.udf_map))
-            elif type(obj) == ResultFile:
-                # TODO: Implement
-                pass
-            else:
-                raise Exception("Unknown type {}".format(obj.type))
-
-        self.session.api.put_batch(update_queue)
+        api_resources = [obj.api_resource for obj in artifacts]
+        self.session.api.put_batch(api_resources)
 
```

Comment:
It is if the domain object has ensured that the underlying resource is updated, which I do in the case of `ResultFile`s and I think all domain objects should do (take care of themselves). This can be seen by running `clarity-ext/test/integration/test_context.py:test_can_update_udfs`.

Is it not being updated in the case of analytes?